docs/source/module_configuration/staticsite.rst

```diff
@@ -78,8 +78,17 @@ Parameters
 
     staticsite_auth_at_edge: true
 
+**staticsite_create_user_pool (Optional[bool])**
+  Creates a User Pool for the Auth@Edge configuration. Either this or ``staticsite_user_pool_arn`` are required when ``staticsite_auth_at_edge`` is ``true``.
+
+  Example:
+
+  .. code-block:: yaml
+
+    staticsite_create_user_pool: true
+
 **staticsite_user_pool_arn (Optional[str])**
-  Required if ``staticsite_auth_at_edge`` is ``true``. A pre-existing Cognito User Pool is required for user authentication.
+  A pre-existing Cognito User Pool is required for user authentication. Either this or ``staticsite_user_pool_arn`` are required when ``staticsite_auth_at_edge`` is ``true``.
```

Comment:
Should be "Either this or ``staticsite_create_user_pool``"

fwiw, you could probably convince me that the need to create a user pool is implicit when an ARN isn't provided, i.e. we could drop the `staticsite_create_user_pool` option and just make it a check for `staticsite_user_pool_arn` being specified. Really not sure what's more logical. I generally prefer explicit configuration, but it can get unwieldy

Reply:
Yeah I thought about that for a while too and was on the fence. I ultimately went with the explicit because I felt it was necessary to make the user fully aware of the decision. If they don't realize they have to create/supply a user pool they may not understand how to populate it with user values.

I'm in the same boat though and could be convinced otherwise.